app/util/fs.py

```diff
@@ -1,5 +1,26 @@
 import os
+import shutil
 import tarfile
+import tempfile
+import subprocess
+
+
+def async_delete(path):
+    """
+    Asynchronously delete a file or a directory. This functionality is handy for deleting large directories.
+    For example, in ClusterRunner, deleting the results directory can take up to half an hour. With async_delete(),
+    the delete call will return almost immediately.
+
+    Under the covers, this is implemented by first synchronously renaming the file or directory to a unique name
+    inside of a temporary directory. After that, we launch an asynchronous process that deletes that file/directory.
+    This asynchronous process does not die when the invoking process (this process) exits/dies.
+
+    :param path: the absolute path to the file or directory to asynchronously delete
+    :type path: str
+    """
+    new_temp_path = tempfile.mkdtemp(prefix='async_delete_directory')
+    shutil.move(path, new_temp_path)
+    subprocess.Popen(['rm', '-rf', new_temp_path])
 
```

Comment:
I forgot we had a blacklist. shutil.move seems like an obvious candidate for something we might forget to mock out, and definitely should.